### ISSUE TITLE
Comparison with pytorch 2.0

### ISSUE THREAD

**Issue by acostin1** [ELS-RD/kernl#197]
Hi folks. 

Not an issue per se but I was looking at kernl to explore better inference for transformer models.

I saw the recent pytorch announcement and their approach with dynamo and inductor and triton seems similar. 

Are there plans to align these two efforts?

**Comment by pommedeterresautee** [comment]
Hi, thank you for your message.
For Bert we have included Inductor in the benchmark (it's the Triton compiler from PyTorch 2.0).
In Inference, PyTorch 2.0 perf are quite far behind (in corner case it may even be slower than doing nothing). 

As [written](https://twitter.com/marksaroufim/status/1599510205058146304) by Mark from PyTorch team, Inductor is thought for training more than inference. In particular it can do something similar to gradient checkpointing in a much more aggressive way. Have a check the PyTorch dev forum for more detailed info.

Moreover, Inductor is generic and optimize kernels based on automatic compiler rules, kernels from this lib are made manually and tend to be much more optimized as we are free to leverage any new paper, etc.

Let me know if it answers you question, or if it needs some clarification.

**Comment by acostin1** [comment]
Thanks for explaining.

I was trying to use it a week ago with FLAN T5 but ran into some errors - I'll give it a try next week and see how much faster it is in BF16 compared with Inductor.

Ty!

**Comment by pommedeterresautee** [comment]
Tks, not tried Flan but if you have some error please report here with reproductible code and we will investigate

**Comment by pommedeterresautee** [comment]
@acostin1 Have you been able to run vanilla PyTorch nightly with Flan ?
It seems that PyTorch 2.0 has an issue with some T5 models... and so we have the same.

**Comment by acostin1** [comment]
I have not - I am running Pytorch latest (with their version of Triton) and it doesn't even compile. Are you folks planning to make Kernl compatible with Pytorch 1.14 nightly?

**Comment by acostin1** [comment]
I'm getting this error when trying the BERT example. Python 3.9.5, pytorch nightly

"name": "BackendCompilerFailed",
	"message": "_compiler raised Exception: Invoking operators with non-Fake Tensor inputs in FakeTensorMode is not yet supported. Please convert all Tensors to FakeTensors first. Found in aten.embedding.default(*(Parameter containing:\ntensor([[-0.0568,  0.0299,  0.0031,  ...,  0.1054,  0.0106, -0.0408],\n        [ 0.0033, -0.0552, -0.0054,  ..., -0.0317, -0.0055, -0.0274],\n        [-0.0132,  0.1356,  0.0669,  ...,  0.0010,  0.0773,  0.0414],\n        ...,\n        [-0.0321,  0.1130,  0.0724,  ...,  0.0187,  0.1477, -0.1466],\n        [-0.1248, -0.1178, -0.0843,  ..., -0.1224,  0.0079, -0.1960],\n        [ 0.0142, -0.0268,  0.0035,  ..., -0.0068,  0.0106, -0.0071]],\n       device='cuda:0', requires_grad=True), FakeTensor(FakeTensor(..., device='meta', size=(1, 8), dtype=torch.int64), cuda:0), 1), **{})\n\nSet torch._dynamo.config.verbose=True for more information\n\n\nYou can suppress this exception and fall back to eager by setting:\n    torch._dynamo.config.suppress_errors = True\n",

**Comment by gaetansnl** [comment]
@acostin1 Hello, Yes we noticed this error on the last nightly. It doesn't happen on the pinned version we have in requirements.txt

**Comment by acostin1** [comment]
I know but we're trying to use best of both worlds ... having you guys fix it will increase the chance of using it.

**Comment by pommedeterresautee** [comment]
FYI, we are switching to a more recent PyTorch nightly: https://github.com/ELS-RD/kernl/pull/220/files

**Comment by pommedeterresautee** [comment]
@acostin1 we have switched to a recent nightly, so you can now have best of both worlds :-)
Please, note that right now PyTorch is a bit buggy. Moreover, the way CUDA graph is implemented has a large memory footprint. We will work on this aspect.
